Standards/scs-0217-v1-baseline-cluster-security.md

```diff
@@ -0,0 +1,141 @@
+
+Most internal clients (like proxies or nodes) are typically authenticated via service accounts or
+x509 certificates, which will normally be created automatically during the setup of a cluster.
+External users can authenticate via an access pattern of choice, which is typically decided by
+the cluster provider.
+
+Authorization is (normally) done by he Role-Based Access Control (RBAC), which matches a request
+by a user with a set of permissions, also called a role. Kubernetes deploys some roles out-of-the-box;
+additional roles need to be carefully checked, since some permissions for specific resources allow
+modification of other resources.
+
+This whole process is especially important for the Kubelet, which allows anonymous requests in its
+default configuration. This is obviously a security risk, since everybody with access to its endpoint
+could manipulate resources that are managed with the Kubelet.
+
+To disable anonymous requests, the Kubelet should be started with `--anonymous-auth=false`.
+Authentication can be provided either through x509 client certificates or API bearer tokens.
+How to set up both approaches can be found in the [Kubelet Authentication and Authorization](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/).
+
+Kubelet authorization is set to `AlwaysAllow` as a default mode. This can be quite problematic,
+since all authenticated users can do all actions. To mitigate this, it is possible to delegate
+authorization to the API server by:
+
+- enabling the `authorization.k8s.io/v1beta1` API group
+- starting the Kubelet with the `--authorization-mode=Webhook` and the `--kubeconfig` flags
+
+After that, the Kubelet calls the `SubjectAccessReview` API in order to determine the authorization of a request.
+
+## Decision
+
+This standard tries to increase security for a Kubernetes cluster in order to provide a
+solid baseline setup with regard to security. For this to work, multiple measures need to be undertaken.
+It is important to note that this standard is not REQUIRED for all clusters,
+but instead gives best practices for increasing security. Nevertheless, if a cluster is claiming
+to be secure, this standard must be applied.
+
+A self-controlled CA SHOULD be used in order to be in control of the TLS certificates, which
+enables the operator to provide and revoke certificates according to the requirements.
+
+All internal endpoints found in the section [TLS for all internal/API traffic] should be
+encrypted with TLS in order to secure internal traffic.
+
+The Kubernetes API (kubeAPI) MUST be secured by authenticating and authorizing the users
+trying to access its endpoints. How a user is authenticated is up to the provider of the
+cluster and/or the wishes of the customer. Authorization MUST be done by providing
+fine-grained RBAC.
+This whole process also MUST be applied to the Kubelet, which in its default configuration
```

Comment:
> This whole process 
Which? The process of authenticating and authorizing?

Reply:
Edited the sentence in https://github.com/SovereignCloudStack/standards/pull/376/commits/5610fe15c9cf2ad9ddf1ddf89f0d575a9002b5d2